### DISCUSSION TITLE
7.1 fixedwing autoland problems

### DISCUSSION THREAD

**Discussion by Sibre3** [iNavFlight/inav#9740] [General]
I have two problems.

The first is with my primary safehome.
I always have a safehome set 300 meters from my launch area. This is because my club does not permit loitering over the pits.
If I place a safehome on the landing strip to only allow landing in a failsafe. nav_rth_allow_landing = failsafe.
It does not give me the option to use my primary safehome 300 meters away, when I select "safehome mode = RTH".

My landing strip is about 2 meters higher at one end than the other. This causes the aeroplane to crash into the ground when approaching from the high side of the strip. And if I set the approach and glide altitude higher to clear it. The aeroplane stalls above the runway.
Is there a way to set the inclination or declination angle of the landing zone on the final approach?

@b14ckyy I don't understand. Isn't this the place where inav release issues are raised? I opened this separate issue report from https://github.com/iNavFlight/inav/issues/9731.

I have experienced two autoland related problems. And one has caused a crash twice. How is autoland any better than the emergency landing procedure? I never would have tested the release candidate if I had known this feature was a beta.
The release notes say nothing about either problem. And it seems like autoland is not compatible with safehome settings?

**Comment by MrD-RC** [comment]
I’ve converted it to a discussion, as these are not issues. The features are working as expected. 

1. There is currently no mechanism for selecting which safehome is used if 2 are within the safehome area. It will use the closest safehome. Safehomes are there for RTH or failsafe. I would suggest for the time being, using the safehome for landing only. If you are using RTH, you are in control. So can exit RTH and loiter at will. If you want to use land on RTH, there is no way around this. 

There is the possibility that in a future release. A safehome could be specified for failsafe only. Which you could set up as a landing strip. Then have another safehome in the area that is only used for RTH, with no landing. But, this would need to be in a new major release. 

2. With the change in height of your landing strip. You would need to use a LiDAR with your aircraft to use autoland. There is no way around that either. The autoland documentation does recommend using a LiDAR for accuracy.

**Comment by b14ckyy** [comment]
2m difference between the high and low side should not cause a crash when properly set up. if you want to allow the landing from both directions, based on wind, then the landing spot should be in the center of the runway. Not at one of the end points. In this case the plane always targets the same point and 1m ground altitude difference does not have an effect as the plane should be in a stable descent slope. 

if you send me the coordinates of the landing area and some details of the plane, I might be able to give you some tips on setting it up properly if the defaults don't work for you.

**Comment by Sibre3** [comment]
Thank you for the reply MrD. I opened this problem under the issues reports. Because that was your recommendation on youtube. I don't understand why it has been move to the back waters, not to be seen. Do you want it to cause more crashes?
I appreciate the clarification you have provided. But I have an issue with this statement.

>The features are working as expected.

How can you say that? It may be working as @Scavanger expected. But it is not working how any unaware person would expect it to.
We are not mind readers. The documentation mentioned nothing of it breaking the safehome functionality. And it only mentions to use the flare function if lidar rangefinder hardware is present.

I don't understand why you both are so eager to hide this issue report? It appears @b14ckyy wrote the documentation. Maybe he doesn't understand the code well enough. I understand oversights in design. But don't try to cover up your failure by blaming me.
The software may be free. But the hardware is not.

**Reply by b14ckyy** [reply]
No one is trying to hide anything. Darren just moved it to the right area. 

The autoland feature works exactly as it was designed to work and as it's documented. So do safe homes.

Autoland is just an extension of safe homes and does not change their behavior in the first place. It is supposed to land at the safe home coordinates. 2 overlapping safe homes were never suggested or advised. 

So you are not reporting an issue but instead you request a new feature. That's different. 

Oh and BTW. The feature was completely specified and planned by me. Scavenger did the coding side to implement it based on my specifications while keeping compatibility with 7.0 configurators.

**Comment by MrD-RC** [comment]
> How can you say that? It may be working as @Scavanger expected. But it is not working how any unaware person would expect it to. We are not mind readers. The documentation mentioned nothing of it breaking the safehome functionality. And it only mentions to use the flare function if lidar rangefinder hardware is present.

It doesn’t break safehome functionality at all. It has always worked so that it will pick the nearest safehome. If you have 2 safehomes set within the allowed safehome radius when you arm. It will pick the nearest. It has always worked this way. 

Autoland is a new feature that builds on safehomes. It has not changed how safehomes are selected. 

LiDAR will add the flare feature. It also means that your aircraft knows how high it is off the ground. This will help if the ground has variation in altitude. But, @b14ckyy may have better suggestions for the settings, so that you don’t need to install one. 

> I don't understand why you both are so eager to hide this issue report? It appears @b14ckyy wrote the documentation. Maybe he doesn't understand the code well enough. I understand oversights in design. But don't try to cover up your failure by blaming me. The software may be free. But the hardware is not.

We are not hiding anything. This is not an issue as it is not a fault or failure with the code. @b14ckyy understands perfectly what autoland is doing. He created the precursor to it using the programming framework in configurator. He has also been working very closely with @Scavanger from the start with this feature. He and Scavenger know more about autoland than anyone else. 

It is not an issue. You are looking for a change from the way it was designed to work. That is more suited to a discussion. We aren’t blaming you. @b14ckyy offered to help tune your settings. He’s trying to help not blame.

**Comment by Sibre3** [comment]
I see the point you are both making. And it is valid.
However, this feature is more suited to a fixed landing location or landing strip. Because of the difficulty in altering the landing coordinates and elevation, anytime we fly somewhere else.. Based on the very slim chance autoland may be required.

> Autoland is just an extension of safe homes and does not change their behavior in the first place
  Autoland is a new feature that builds on safehomes. It has not changed how safehomes are selected.

I'm not sure if you're aware of model flying club regulations. Surrounding no flying or loitering over the pit area. _Which is what autoland does to establish the wind direction._
Can either of you explain how I can set `nav_rth_allow_landing = failsafe.` and not have it attempt to autoland at my primary safehome in the middle of an empty field. Without it assuming the untouched default approach elevation settings are correct for that location?
In previous releases, the aeroplane would perform an emergency loiter to ground level, while reducing the throttle. How do I now achieve this in inav 7.1? 
I consider it just a safe to use the old landing method in some cases, since autoland does not terrain map.

>if you want to allow the landing from both directions, based on wind, then the landing spot should be in the center of the runway

All this may be true. But how was I to know this from the document, before you told me here. 
Why wouldn't I assume an _easy to use_ autoland feature would not assess the approximate rise and fall of the landing terrain from bing maps. And apply that to the landing approach zone?
As a lowly user I had no clue this feature was so basic, unless you added this fact to the documents.

**Comment by b14ckyy** [comment]
Everything autoland can do is in the documentation. If something is not nescribed there, it's not done.

Anyway. We can continue to discuss how simple and basic autoland is in your opinion or we can help you to find proper settings to make it work.

Yes the loiter above the landing location to verify the wind direction and speed is mandatory at the moment. This might change in future updates.

If you don't define landing approaches for your safe home location, the classic circle land is used.

As I said earlier, send me the coordinates of your field where to land and I can have a look and maybe figure out a proper landing option.

**Comment by MrD-RC** [comment]
> I see the point you are both making. And it is valid. However, this feature is more suited to a fixed landing location or landing strip. Because of the difficulty in altering the landing coordinates and elevation, anytime we fly somewhere else.. Based on the very slim chance autoland may be required.

This feature **only**  works with fixed landing locations. It can only be used with a safehome or, I believe, a waypoint. So the location and approach vectors will always be fixed. Automatic landing elsewhere will default back to the old loiter until touchdown method. Which again, doesn’t know the height of the terrain. So just uses the 0 defined at arming. 

> I'm not sure if you're aware of model flying club regulations. Surrounding no flying or loitering over the pit area. _Which is what autoland does to establish the wind direction._ Can either of you explain how I can set `nav_rth_allow_landing = failsafe.` and not have it attempt to autoland at my primary safehome in the middle of an empty field. Without it assuming the untouched default approach elevation settings are correct for that location? In previous releases, the aeroplane would perform an emergency loiter to ground level, while reducing the throttle. How do I now achieve this in inav 7.1? I consider it just a safe to use the old landing method in some cases, since autoland does not terrain map.

As I have said previously. It is not possible to use 2 different safehomes in the same area. It will always use the closest. Any changes to this would need to wait til the next major release. As it would break backwards compatibility. Also, the only thing that could be changed is to have one safehome for failsafe, and another for regular RTH. It couldn't be any more than that. 

As for club rules. They were taken into consideration. That is the reason you can choose the base leg side on the approach. So that it won’t fly over out of bounds area. However, it does seem like the loiter should be taking place offset from the landing area. In the direction of the approach pattern. But please note, this is still not a bug or issue. It is a request for a change to the feature. It is still working as intended at this point in time. 

> All this may be true. But how was I to know this from the document, before you told me here. Why wouldn't I assume an _easy to use_ autoland feature would not assess the approximate rise and fall of the landing terrain from bing maps. And apply that to the landing approach zone? As a lowly user I had no clue this feature was so basic, unless you added this fact to the documents.

It could be made more obvious in the document. But to me, if you can approach from both directions. It stands to reason that the landing point has to accommodate both. So should be in the middle of the landing area.

Where is it stated that autoland is _easy to use_? It is _advanced automatic landing for fixed wing_. It requires the aircraft and navigation to be well tuned. Plus the settings for the landing itself be tweaked for the aircraft and environment. However, it is easier and more feature rich than with other autopilots. Where you don’t have multiple approaches based on wind, and have to set up the approach manually each time using waypoints. 

Bing! Maps doesn’t have very accurate altitude data. We actually tested this and found it was out by quite a bit. This actually lead to a change in Mission Control, to give more accurate terrain altitude. But, it can still be out. Also, the altitude can drift. It is not uncommon drift to occur with the barometer. So even knowing the exact terrain altitude, it could still be incorrect due to drift. For the best automatic landing performance. A LiDAR should be used.



I think the loiter position could be addressed. There is a valid reason for not performing the wind estimation loiter directly over the safehome. But, I also believe it should be an option. For those not flying in a club environment. There’s no reason to over complicate things.

**Comment by b14ckyy** [comment]
@Scavanger Maybe a good point for optimization in the future. Instead of loitering directly around the safe home, move the loiter point in the approach direction to have it offset by loiter radius based on the left/right setting. 

All the fields I have flown so far have landing areas far enough away from the pits or at least the pits are at the end of the strip so circling over the center was never an issue.

**Comment by kasatka60** [comment]
How often does the wind direction and speed change? Is it unsafe to use data on direction and speed 5-10 minutes ago when approaching?

**Reply by MrD-RC** [reply]
It loiters just before landing to check the wind direction.

**Reply by sensei-hacker** [reply]
Depends on the local geography. If you're flying next to a very large building like I do, it changes over the course of above 50 feet. :)

Also depends on where the front is - I'm sure you've noticed the wind suddenly pick up and reverse direction over the course of five minutes or so. That's when the front is passing.

**Reply by MrD-RC** [reply]
Over 5 minutes, sure. But autoland checks the wind direction, then flies out to base. It lands within a minute or two after checking the wind.

**Comment by kasatka60** [comment]
![1](https://github.com/iNavFlight/inav/assets/61105679/20873edb-bfa8-4b38-b136-9103775cb070)
![21](https://github.com/iNavFlight/inav/assets/61105679/eb635122-e978-4f0e-956f-1cf7c3227d9a)
https://youtu.be/x022m_YDKuE
I have tested one flight so far.
The blue line is how you should approach for landing.
Green line - real flight.
Does it make sense to move the point so that plane doesn’t fly over when turning (red line).
Or is it not safe?

**Reply by b14ckyy** [reply]
There seems to be a pretty severe tuning issue with your navigation system. the plane really struggles to do the turns at all and with the tailwind it overshoots completely. 
One thing that will change is the default Waypoint hit-radius. Check advanced tuning. The WP Radius is 50m by default up to RC1. Change that to 10m on a wing and 8m if its a plane with tail. 
Then when flying in these kind of windy conditions you need to raise the maximum navigation bank angle. I don't know your current settings but it looks like the bank angle is limited to something like 30°. Raise that up to at least 45°, better something in the 50-60° ballpark for navigation (Wings need more than planes with tail). Then make sure the Angle Limit in the PID Tuning page under Mechanical settings is the same or higher than that. This should allow the plane to make these tail wind turns tighter. 

Do you have WP Path tracking turned on? It is not mandatory but if you have a tight landing corridor its highly recommended especially for windy conditions. this will keep the plane better on track. You can turn it on with a tracking accuracy of 6 to start with. Its in Advanced Tuning. 

approach length is more than enough, speed looks okay, altitude as well during approach. But maybe consider to set the Glide Altitude 3-4m lower as the motor cut a bit too early and the plane nearly stalled. So land alt at 4600 instead of 5000.

**Reply by kasatka60** [reply]
Thanks for the tips. You almost got it right with the settings. I will change and check. Wing z-84

**Reply by kasatka60** [reply]
Configured according to your recommendations. Now the plane lands perfectly.

**Reply by b14ckyy** [reply]
Glad to hear! Maybe @Choolet should have a look on this as well. For now I will close this thread.

**Comment by Sibre3** [comment]
My landing strip is 40m wide. And I conducted my flights with very little wind. 
During these consecutive trials I had more than 20 satellites. HDOP was less than 1.3. I had GPS and Galileo selected.

The issue is I can not get auto land to work the same each time I use it. 
I conducted 22 landing attempts, one after another. With only 5 of them managing to get the aeroplane aligned correctly within the 40m landing strip.
They were mostly off the right side of the landing strip. And if I moved the safehome farther to the left of the landing strip. It made a difference once, then not again after.
I attempted to plot a mission over my landing strip, to see if it was GPS inaccuracy causing issues. But it tracked perfectly down the landing strip each time I restarted the mission. With the heading being relatively precise.


@Scavanger Can you please elaborate more on operation of `nav_fw_land_final_approach_pitch2throttle_mod`

```
If the parameter is set to 200% and Pitch To Throttle Ratio is set to 15,
Pitch To Throttle Ratio is set to 30 on the final approach. This causes a reduction in engine
power on approach when the nose is pointing downwards
```

I have observed discrepancies in landing speed. Mostly it attempts to land at flying speed. More than 60km/h.
When I increase `nav_fw_land_final_approach_pitch2throttle_mod` incrementally from 100 to 400%. I could not see any significant reduction in throttle on the landing approach. 
I have tried to raise `nav_fw_land_glide_pitch` to slow the aeroplane, with mixed results. 
Does `nav_fw_land_final_approach_pitch2throttle_mod` remain active in the Glide phase, as well as the Final approach?

What are the hard coded distances for `Final approach` and `Glide approach` as a percentage of the `nav_fw_land_approach_length`.
There are many variables that need to be accounted for here.
* Not all aeroplanes can slow down within the same distance.
* Some glide faster that others.
* Lighter aeroplanes have less penetration into a stronger headwind. Requiring more throttle to even make it too the Flare location.

I'm sorry if I come across sounding negative. Its seems this feature could have been done better. By accounting for the realistic variables model aircraft will inevitably encounter. After wasting practically a whole day tuning.  I can't say this feature gives me much more confidence than the normal emergency landing circle method. Which is a real shame.

**Reply by b14ckyy** [reply]
Hi Choolet, thanks for your questions, quite a bit to answer :D 

> I conducted 22 landing attempts, one after another. With only 5 of them managing to get the aeroplane aligned correctly within the 40m landing strip.

You have probably not yet enabled and tuned the Waypoint Path Tracking feature. Its in advanced tuning. This feature, since 6.0, enables INAV to not just fly towards the next WP in a mission but actually align to the connecting lines between waypoint and follow a specific path in space. We decided to not enable it forcefully for Autoland (like we did with turn_smoothing = cut) because it might need separate tuning and not for everyone it will be necessary. 

Another issue is the Default Waypoint trigger radius in INAV since 6.0 I think. Its too big in the old presets and will cause the plane to turn too early. Please apply the following settings (these will be in the final release notes): 

Plane with no tail: `set nav_wp_radius = 1000`

Plane with tail: `set nav_wp_radius = 800`

------------------------------------

> Can you please elaborate more on operation of `nav_fw_land_final_approach_pitch2throttle_mod`

I will Update that part a bit. But as a short simplification (assuming you know how Pitch2Throttle already works):
If the nav system pitches the nose down below horizon, the pitch to throttle will be boosted by the `nav_fw_land_final_approach_pitch2throttle_mod` percentage. If your P2T is globally set to 20 and the boost value is set to 150%, then your P2T during descent will be 30. If the nose pitches up above horizon, this boost is disable to allow a normal and stable climb without overshooting. 

The P2T boost is just an interim solution until INAV gets a better autonomous throttle management system for NAV modes. the problem with the current solution is, that especially very aerodynamic planes can speed up quite a bit when descending with throttle on. Adjusting the throttle linearily based on pitch angle in both directions is not a good solution and we needed to allow pilots to have a much lower throttle during approach, than they would have during navigation flights. 

In short: It slows down your plane for descent but still prevents a stall during climb if necessary. You need to test it out how much is needed if at all. 

> When I increase nav_fw_land_final_approach_pitch2throttle_mod incrementally from 100 to 400%. I could not see any significant reduction in throttle on the landing approach.

In this case, your approach is too steep. Either make the approach longer globally or reduce the approach height to have a shallower approach. The glide ratio of your plane probably overcomes the throttle reduction in this case. The Glide only is there to reduce the vertical speed in time before hitting the ground. Its not affected anymore by the throttle mod as the motor is cut when glide starts.

--------------------------------------

> What are the hard coded distances for Final approach and Glide approach as a percentage of the `nav_fw_land_approach_length`.

There are no hardcoded values. The final approach is as long as it needs to be until Glide altitude is triggered. `nav_fw_land_approach_length` is the full distance between the Landing waypoint (At Safe Home Location and at Land Altitude, so a point in space, not on the ground) and the final approach turn waypoint. The distances when Glide and Flare are executed depend in the trigger altitude and the approach slope that results from approach length and approach altitude. 

> Not all aeroplanes can slow down within the same distance.

Yes, that's why approach length is globally set for all landings to give the plane the necessary distance to align and slow down a bit. Also that's why Glide alt, Glide angle and Flare alt and angle are global settings as they are always the same for a specific build

> Some glide faster that others.

That's why you have to select the Approach altitude accordingly. based on approach altitude and the approach length you can easily figure out the expected descent ratio and compare it with your glide ratio of the craft. That's why we start with default values of 30m and 350m. That results roughly in a 1:10 glide ratio and was a compromise between out of box compatibility with many planes and a not too crazy distant approach. In the end its up to the pilot to tune. 

> Lighter aeroplanes have less penetration into a stronger headwind. Requiring more throttle to even make it too the Flare location.
INAV has a minimum ground speed limit that can be configured by `nav_min_ground_speed` from 6-50m/s. Also the way inav manages altitude changes between waypoints keeps the plane on track vertically and it will constantly adjust the descent speed to arrive at the same altitude every time. 

---------------------------------------

Regarding your last statement. This is a initial iteration of Autoland and it had specific requirements to met for a commercial use feature request it was initially made for. Of course we expected to run into edge cases where some advancements are needed and these will be considered in future versions. As an example, a way to off-set the Wind-Measurement loiter will be needed for flying clubs that don't allow flying over the bunkers close to the landing strip. These are things that need to be figured out on the go and addressed with updates. 

Your main issue was very likely (same for @kasatka60 above) that path tracking as off and waypoint radius was too big. This is indeed a hole in the documentation and will be changed. If you fix these 2 issues, you will probably have a 5x better experience.

**Reply by b14ckyy** [reply]
I updated the Autland Documentation with General Parameters and Tuning Tips: 
https://github.com/iNavFlight/inav/blob/491680a0fb5741b252ba460e7969e9b35083133a/docs/Fixed%20Wing%20Landing.md

**Reply by Sibre3** [reply]
Hello. Thanks for replying.

>You have probably not yet enabled and tuned the Waypoint Path Tracking feature.

I have those settings enabled and I tested them with various values, with little change to the theme.
```
nav_wp_radius = 500
nav_fw_wp_tracking_accuracy = 4
nav_fw_wp_tracking_max_angle = 55
nav_fw_wp_turn_smoothing = ON-CUT
```
Did you see this part I wrote.  ↓↓↓  Waypoint accuracy is good.

>I attempted to plot a mission over my landing strip, to see if it was GPS inaccuracy causing issues. But it tracked perfectly down the landing strip each time I restarted the mission. With the heading being relatively precise.

This proves something else is wrong with the land approach heading. I have notice that when a RIGHT turn is made. The aeroplane tries to land 10-15m to the right of the safehome and the same applies, but opposite for a LEFT turn.

>But as a short simplification (assuming you know how Pitch2Throttle already works.

I am aware how pitch2thr works. Although after taking it up to 400% and seeing little difference. I can only conclude something isn't right with it.
These are my settings.

```
nav_fw_land_approach_length = 400
nav_fw_land_final_approach_pitch2throttle_mod = 250
nav_fw_land_glide_alt = 180
nav_fw_land_flare_alt = 80
nav_fw_land_glide_pitch = 1
nav_fw_dive_angle = 16
nav_fw_pitch2thr = 9
nav_fw_pitch2thr_threshold = 50
```

>That's why you have to select the Approach altitude accordingly. based on approach altitude and the approach length you can easily figure out the expected descent ratio and compare it with your glide ratio of the craft. That's why we start with default values of 30m and 350m. That results roughly in a 1:10 glide ratio and was a compromise between out of box compatibility with many planes and a not too crazy distant approach.

No offense. But I even find setting my approach distance to 400m with and approach altitude of 40m as crazy.
I can't go any lower than 40m, to clear trees within the safehome loiter area, required to get a wind heading.

Getting back to pitch2thr. Does `nav_fw_land_final_approach_pitch2throttle_mod ` account for `nav_fw_pitch2thr_threshold`?
Meaning the aeroplane would require a dive angle greater than -5° before pitch2thr or negative boost would become operable.
This may explain why I never seen 400% reduce the throttle. Because the pitch approach angle never went below -5° once in the `nav_fw_land_approach_length` distance.

And don't overlook pitch2thr is constrained by `nav_fw_dive_angle`. And the micro second throttle value between `nav_fw_cruise_thr ` and  `nav_fw_min_thr`.

**Reply by b14ckyy** [reply]
Can you provide a flight blackbox log of one of the attempts please? Its better If I can have a look. 
Maybe we overlook something. Also a complete diff of your setup. If you don't wanna share any location, you can send it over to me on FB (Marc Hoffmann in the INAV groups) or discord (blackyy). 

> Getting back to pitch2thr. Does nav_fw_land_final_approach_pitch2throttle_mod  account for nav_fw_pitch2thr_threshold?
> Meaning the aeroplane would require a dive angle greater than -5° before pitch2thr or negative boost would become operable.
> This may explain why I never seen 400% reduce the throttle. Because the pitch approach angle never went below -5° once in the nav_fw_land_approach_length distance.

This is indeed a valid point and should be optimized in the future. For this first implementation though, we had to use what's available as throttle control for the time being. But I don't think that the throttle control, if the plane still flies at relatively modest 60kph, has a big effect on the approach direction for you. We have tested this on planes that approached faster than 60 and with approach length down to 300m with very precise landings. But checking a flight log might give some clues what's wrong. 

> And don't overlook pitch2thr is constrained by nav_fw_dive_angle. And the micro second throttle value between nav_fw_cruise_thr  and nav_fw_min_thr.

This is correct and of course needs to be set lower if it prevents the plane from slowing down in general. I set the min throttle on all my planes to 1000 usually because they reach a cruise speed glide angle usually way before the motor stops. On very powerful setups with high glide ratio and high pitch props, leaving the motor running even at slow speeds can prevent the plane from slowing down at all.

**Reply by Sibre3** [reply]
I don't have any logs because one aeroplane has an F411 without logging. And the other has a H743, who's card logging doesn't work with iNAV. But does with Ardupilot.

I'm certain there's nothing wrong with my settings. Both aircraft follow missions with good accuracy.
I've flown iNAV for many years. I don't believe that I've set it up incorrectly.

   >This is indeed a valid point and should be optimized in the future

You summed it up well with that statement. It needs a lot of optimizing.

I found this video on Youtube https://youtu.be/2ZbU2zjXDmc?si=1YpWRtgjShFP2LVn&t=119 
 It shows exactly what I described. It seems unlikely he set the safehome in the gutter on the right side of the road.
All the other tests I have seen of the feature, have been done in the middle of a field. So the inaccuracy goes unnoticed.

Anyway, I'm over it. Hopefully it gets fixed one day. Thanks for your time.

**Reply by b14ckyy** [reply]
The video you showed was a test flight of one of the people who tested the feature during development. And Yes he indeed set the landing location on the side of the road. He landed within 2m side to side on the landing spot. 

All I can say is, that all our tests so far have shown perfect landing behavior when set up correctly. 

And btw, the H743 logging works. You either need to put an empty 4GB or maybe 8GB card, fleshly FAT32 formatted, into the FC power it via USB and let it sit there for 10-20min and it should work. Or you "prepare" it in a F405, F722 or F765 FC by putting it in there for the first time and power it up for 30s. Then you can move it to the H7 and it should work as well. 

Edit: you can also collect OpenTX or EdgeTX telemetry logs that I can also use to plot the course on a map and compare it with your home settings.

**Reply by fatbaldmen** [reply]
@Choolet 

That is my video you referenced to, and bl4ckyy is right, that is precisely how I set the landing to be, and the accuracy was less than 2m. Bl4ckyy is head of the testers who is in direct comms with the dev, and all the testers info and videos go through him. In our early testing stages, Marc had compared a landing with my OTX logs and found the accuracy there, to be less than 2m, closer to 1m. Join the FB group inavfixedwinggroup, some of us can help get you sorted or find what any issues may be 🍻

**Reply by Sibre3** [reply]
@b14ckyy I discovered the cause of the heading alignment problem.
On a certain flight the auto landing altitude drifted. Causing the aeroplane to touch down with the motor still spinning. This caused the propeller to get slightly damaged. Enough to cause some vibrations. Which lead to AHRS drift.
All of this took place after we were recommended in this video  https://www.youtube.com/watch?v=PTY_fCa_aQE to increase the gyro_main_lpf_hz.
Once I reduced the filter back to default. And after confirming it was filter related I later replace the slightly damaged propeller. Auto landing now works as expected.
Thank you for your patients.

**Reply by b14ckyy** [reply]
Glad to hear and happy landings :)
